### PR TITLE
Let users run actions from the model detail page

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/28305

Changes:
- Removes eye/edit button and replaces it with a button to run the action
- It opens a model to input all the required parameters

Screenshot 2023-02-17 at 15 57 45
Screenshot 2023-02-17 at 15 57 54
Screenshot 2023-02-17 at 15 58 08
Screenshot 2023-02-16 at 22 36 13

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28361)

### CODE REVIEW

File: frontend/src/metabase/actions/utils.ts [hunk 1]
```diff
@@ -211,3 +214,44 @@ export const isActionCard = (card: Card) => card?.display === "action";
 export const getFormTitle = (action: WritebackAction): string => {
   return action.visualization_settings?.name || action.name || t`Action form`;
 };
+
+function hasDataFromExplicitAction(result: any) {
```
Comment (ranquild) [inline frontend/src/metabase/actions/utils.ts:218]: Moved from dashboards

File: frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+type ActionExecuteModalProps = OwnProps & ActionLoaderProps;
+
+const ActionExecuteModal = ({
+  action,
+  onSubmit,
+  onClose,
+}: ActionExecuteModalProps) => {
+  const title =
+    action.parameters.length > 0 ? action.name : t`Run ${action.name}?`;
```
Comment (kulyk) [inline frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx:35]: I wonder if we can tweak / use [`getFormTitle` from `metabase/actions/utils`](https://github.com/metabase/metabase/blob/c7a19a4040c2ee3fff1e669d77a61764f7a6d1c5/frontend/src/metabase/actions/utils.ts#L223) here somehow?

File: frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      />
+    </ModalContent>
+  );
+};
+
+const mapDispatchToProps = {
+  onSubmit: executeAction,
+};
+
+export default _.compose(
```
Comment (kulyk) [inline frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx:60]: nit: move `mapDispatchToProps` above the component?

File: frontend/src/metabase/actions/utils.ts (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/actions/utils.ts:243]: Awesome! 👏

File: frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionListItem.tsx [hunk 4]
```diff
@@ -44,77 +46,73 @@ function ImplicitActionCardContent() {
...
-    // Remove this optional chaining after removed all the existing actions without creators
-    action.creator?.common_name && t`Created by ${action.creator.common_name}`,
-  ].filter(isNotNull);
-
-  const hasMenu = menuItems.length > 0;
+  const menuItems = useMemo(
+    () => [
+      {
+        title: canWrite ? t`Edit` : t`View`,
+        icon: canWrite ? `pencil` : `eye`,
```
Comment (kulyk) [inline frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionListItem.tsx:69]: nit: let's not use backticks when we don't need a template string

**Review by kulyk — APPROVED** [review]
Overall looking and working great! Left a few small rather non-blocking comments

File: frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/actions/containers/ActionExecuteModal/ActionExecuteModal.tsx]: Fixed

File: frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionListItem.tsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionListItem.tsx]: Fixed

**Pull request merged**